statistics/ddl.go

```diff
@@ -98,42 +103,52 @@ func (h *Handle) insertColStats2KV(tableID int64, colInfo *model.ColumnInfo) err
 			defer terror.Call(rs[0].Close)
 		}
 		if err != nil {
-			return errors.Trace(err)
+			return
 		}
 		chk := rs[0].NewChunk()
 		err = rs[0].Next(ctx, chk)
 		if err != nil {
-			return errors.Trace(err)
+			return
 		}
 		count := chk.GetRow(0).GetInt64(0)
 		value := types.NewDatum(colInfo.OriginDefaultValue)
 		value, err = value.ConvertTo(h.mu.ctx.GetSessionVars().StmtCtx, &colInfo.FieldType)
 		if err != nil {
-			return errors.Trace(err)
+			return
 		}
 		if value.IsNull() {
 			// If the adding column has default value null, all the existing rows have null value on the newly added column.
 			_, err = exec.Execute(ctx, fmt.Sprintf("insert into mysql.stats_histograms (version, table_id, is_index, hist_id, distinct_count, null_count) values (%d, %d, 0, %d, 0, %d)", h.mu.ctx.Txn().StartTS(), tableID, colInfo.ID, count))
 			if err != nil {
-				return errors.Trace(err)
+				return
 			}
 		} else {
 			// If this stats exists, we insert histogram meta first, the distinct_count will always be one.
 			_, err = exec.Execute(ctx, fmt.Sprintf("insert into mysql.stats_histograms (version, table_id, is_index, hist_id, distinct_count, tot_col_size) values (%d, %d, 0, %d, 1, %d)", h.mu.ctx.Txn().StartTS(), tableID, colInfo.ID, int64(len(value.GetBytes()))*count))
 			if err != nil {
-				return errors.Trace(err)
+				return
 			}
 			value, err = value.ConvertTo(h.mu.ctx.GetSessionVars().StmtCtx, types.NewFieldType(mysql.TypeBlob))
 			if err != nil {
-				return errors.Trace(err)
+				return
 			}
 			// There must be only one bucket for this new column and the value is the default value.
 			_, err = exec.Execute(ctx, fmt.Sprintf("insert into mysql.stats_buckets (table_id, is_index, hist_id, bucket_id, repeats, count, lower_bound, upper_bound) values (%d, 0, %d, 0, %d, %d, X'%X', X'%X')", tableID, colInfo.ID, count, count, value.GetBytes(), value.GetBytes()))
 			if err != nil {
-				return errors.Trace(err)
+				return
 			}
 		}
 	}
-	_, err = exec.Execute(ctx, "commit")
+	return
+}
+
+// Commit when error is nil, otherwise rollback.
+func doneTransaction(ctx context.Context, exec sqlexec.SQLExecutor, err error) error {
```

Comment:
`finishTransaction` is better.
And the comment should starts with the function name.